extension-manifest-v3/src/background/adblocker.js

```diff
@@ -391,21 +391,21 @@ async function injectScriptlets(tabId, url) {
   }
 
   if (scriptlets.length > 0) {
-    executeScriptlets(tabId, scriptlets);
+    executeScriptlets(tabId, frameId, scriptlets);
   }
 }
 
 if (__PLATFORM__ === 'safari') {
   chrome.runtime.onMessage.addListener((msg, sender) => {
     if (sender.url && msg.action === 'injectScriptlets') {
-      injectScriptlets(sender.tab.id, sender.url);
+      injectScriptlets(sender.tab.id, sender.frameId, sender.url);
```

Comment:
I really don't expect the case that Safari doesn't give `frameId` because we're always accessing to tab information. However, it would be nice if we can measure how it'll be important to fallback with zero value (means main_frame in Safari).